receiver/prometheusreceiver/metrics_receiver_test.go

```diff
@@ -1062,3 +1062,81 @@ func testEndToEnd(t *testing.T, targets []*testData, useStartTimeMetric bool) {
 		target.validateFunc(t, target, results[target.name])
 	}
 }
+
+var startTimeMetricRegexPage = `
+# HELP go_threads Number of OS threads created
+# TYPE go_threads gauge
+go_threads 19
+# HELP http_requests_total The total number of HTTP requests.
+# TYPE http_requests_total counter
+http_requests_total{method="post",code="200"} 100
+http_requests_total{method="post",code="400"} 5
+# HELP http_request_duration_seconds A histogram of the request duration.
+# TYPE http_request_duration_seconds histogram
+http_request_duration_seconds_bucket{le="0.05"} 1000
+http_request_duration_seconds_bucket{le="0.5"} 1500
+http_request_duration_seconds_bucket{le="1"} 2000
+http_request_duration_seconds_bucket{le="+Inf"} 2500
+http_request_duration_seconds_sum 5000
+http_request_duration_seconds_count 2500
+# HELP rpc_duration_seconds A summary of the RPC duration in seconds.
+# TYPE rpc_duration_seconds summary
+rpc_duration_seconds{quantile="0.01"} 1
+rpc_duration_seconds{quantile="0.9"} 5
+rpc_duration_seconds{quantile="0.99"} 8
+rpc_duration_seconds_sum 5000
+rpc_duration_seconds_count 1000
+# HELP example_process_start_time_seconds Start time of the process since unix epoch in seconds.
+# TYPE example_process_start_time_seconds gauge
+example_process_start_time_seconds 400.8
+`
+
+// TestStartTimeMetricRegex validates that timeseries have start time regex set to 'process_start_time_seconds'
+func TestStartTimeMetricRegex(t *testing.T) {
+	targets := []*testData{
+		{
+			name: "target1",
+			pages: []mockPrometheusResponse{
+				{code: 200, data: startTimeMetricRegexPage},
+			},
+			validateFunc: verifyStartTimeMetricPage,
+		},
+	}
+	testEndToEndRegex(t, targets, true, ".*_process_start_time_seconds$")
```

Comment:
As mentioned in the other comment the leading underscore won't match the default name.
Could you actually add a test scenario where regex is `.*process_start_time_seconds$` and the metrics is `process_start_time_seconds` (i.e. no prefix)?
i.e. run this test with `startTimeMetricPage`.

Reply:
Done.